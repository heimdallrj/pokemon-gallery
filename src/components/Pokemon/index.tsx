import { Container, Title, Details, Row, Attr, Avatar } from './styled';

type Props = {
  name: string;
  avatar?: string;
  weight?: null | string;
  abilities?: string[];
  moves?: string[];
};

export default function Pokemon({
  name,
  avatar,
  weight = null,
  abilities = [],
  moves = [],
}: Props) {
  return (
    <Container>
      <Title>{name}</Title>

      {avatar && <Avatar src={avatar} alt={name} />}

      <Details>
        {weight && (
          <Row>
            <Attr>Weight:</Attr> {weight}
          </Row>
        )}
        {abilities.length > 0 && (
          <Row>
            <Attr>Abilities:</Attr> {abilities.join(', ')}
          </Row>
        )}
        {moves.length > 0 && (
          <Row>
            <Attr>Moves:</Attr> {moves.join(', ')}
          </Row>
        )}
      </Details>
    </Container>
  );
}
