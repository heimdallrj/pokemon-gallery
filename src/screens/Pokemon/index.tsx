import AppLayout from 'layouts/AppLayout';

export default function Pokemon() {
  return <AppLayout>Pokemon</AppLayout>;
}
