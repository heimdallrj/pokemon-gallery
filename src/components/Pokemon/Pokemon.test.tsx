import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';

import Pokemon from './index';

const pokemon = {
  name: 'venonat',
  avatar:
    'https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/48.png',
  weight: '18',
  abilities: ['a', 'b', 'c'],
  moves: ['x', 'y'],
};

describe('Components - Pokemon', () => {
  test('renders when only name has provided', () => {
    const component = <Pokemon name={pokemon.name} />;
    render(component);

    expect(screen.getByText(/venonat/i)).toBeInTheDocument();

    // Snapshot
    const tree = renderer.create(component).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders when all the props has provided', () => {
    const component = (
      <Pokemon
        name={pokemon.name}
        avatar={pokemon.avatar}
        weight={pokemon.weight}
        abilities={pokemon.abilities}
        moves={pokemon.moves}
      />
    );
    render(component);

    expect(screen.getByText(/venonat/i)).toBeInTheDocument();
    expect(screen.getByText(/Weight:/i)).toBeInTheDocument();
    expect(screen.getByText(/Abilities:/i)).toBeInTheDocument();
    expect(screen.getByText(/Moves:/i)).toBeInTheDocument();

    // Snapshot
    const tree = renderer.create(component).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
