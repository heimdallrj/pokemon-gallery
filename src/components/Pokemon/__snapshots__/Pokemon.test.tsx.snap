// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components - Pokemon renders when all the props has provided 1`] = `
<div
  className="sc-bdvvtL"
>
  <h2
    className="sc-gsDKAQ kGDybl"
  >
    venonat
  </h2>
  <img
    alt="venonat"
    className="sc-jRQBWg blClLs"
    src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/48.png"
  />
  <div
    className="sc-dkPtRN jNTvLF"
  >
    <div
      className="sc-hKwDye cNjaZt"
    >
      <span
        className="sc-eCImPb eMNjgV"
      >
        Weight:
      </span>
       
      18
    </div>
    <div
      className="sc-hKwDye cNjaZt"
    >
      <span
        className="sc-eCImPb eMNjgV"
      >
        Abilities:
      </span>
       
      a, b, c
    </div>
    <div
      className="sc-hKwDye cNjaZt"
    >
      <span
        className="sc-eCImPb eMNjgV"
      >
        Moves:
      </span>
       
      x, y
    </div>
  </div>
</div>
`;

exports[`Components - Pokemon renders when only name has provided 1`] = `
<div
  className="sc-bdvvtL"
>
  <h2
    className="sc-gsDKAQ kGDybl"
  >
    venonat
  </h2>
  <div
    className="sc-dkPtRN jNTvLF"
  />
</div>
`;
